jobs/src/main/scala/dcos/metronome/queue/QueueService.scala

```diff
@@ -0,0 +1,18 @@
+package dcos.metronome.queue
+
+import mesosphere.marathon.core.launchqueue.LaunchQueue.QueuedTaskInfo
+
+/**
+  * Provides access to the underlying list of tasks in the launch queue.
+  *
+  */
+trait QueueService {
+
+  def list(): Iterable[QueuedTaskInfo]
```

Comment:
I would personally prefer to create our own domain objects with subset of Marathon's QueuedTaskInfo and RunSpec and bind to those on this layer.

I don't feel that well about passing internal Marathon objects throughout the whole project since those might change in the future version and I feel it might be valuable to isolate places where the changes would be necessary to a small layer that is transforming Marathon internal stuff to our own stuff...

Although I feel kind of strongly about this it is definitely up to discussion! :)

Reply:
as always it depends :) 

if we are targeting a future merge of metronome and marathon than creating a bunch of types that need refactoring every time marathon changes seems like creating work.   In general you and I have similar views.